src/consensus/aft/raft.h

```diff
@@ -1103,26 +1106,49 @@ namespace aft
           return;
         }
 
-        state->last_idx = i;
-
-        Term sig_term = 0;
-        Index sig_index = 0;
-        auto tx = store->create_tx();
-        kv::DeserialiseSuccess deserialise_success;
-        ccf::PrimarySignature sig;
-        kv::ConsensusHookPtrs hooks;
-        if (consensus_type == ConsensusType::BFT)
+        auto ds = store->deserialise(entry, consensus_type, public_only);
+        if (ds == nullptr)
         {
-          deserialise_success = store->deserialise_views(
-            entry, hooks, public_only, &sig_term, &sig_index, &tx, &sig);
+          LOG_DEBUG_FMT("failed to deserialize we failed to apply");
+          send_append_entries_response(
+            r.from_node, AppendEntriesResponseType::FAIL);
+          return;
         }
-        else
+        append_entries.push_back(std::make_tuple(std::move(ds), i));
+      }
+
+      for (auto& ae : append_entries)
+      {
+        auto& [ds, i] = ae;
+        state->last_idx = i;
+
+        kv::DeserialiseSuccess deserialise_success =
+          kv::DeserialiseSuccess::FAILED;
+        if (ds != nullptr)
         {
-          deserialise_success =
-            store->deserialise(entry, hooks, public_only, &sig_term);
+          deserialise_success = ds->Execute();
+          if (deserialise_success == kv::DeserialiseSuccess::FAILED)
+          {
+            state->last_idx--;
+            ledger->truncate(state->last_idx);
+
+            commit_if_possible(state->last_idx);
+
+            // The term may have changed, and we have not have seen a signature
+            // yet.
+            auto lci = last_committable_index();
+            if (r.term_of_idx == aft::ViewHistory::InvalidView)
```

Comment:
Is this now duplicated? Also, we don't have a warning for this until clang 10, but I think we're trying to avoid meaningful indentation.